tests/basic/async_dr_test.go

```diff
@@ -376,3 +690,135 @@ func DeleteAndWaitForMigrationDeletion(name, namespace string) error {
+			contexts = append(contexts, ScheduleApplications(taskName)...)
+		}
+	}
+	for _, ctx := range contexts {
+		// Override default App readiness time out of 5 mins with 10 mins
+		ctx.ReadinessTimeout = appReadinessTimeout
+		namespace := GetAppNamespace(ctx, "")
+		migrationNamespaces = append(migrationNamespaces, namespace)
+	}
+	log.Infof("Migration Namespaces are : [%v]", migrationNamespaces)
+	ValidateApplications(contexts)
+	return migrationNamespaces, contexts
+}
+
+func createMigSchdAndValidateMigration(migSchedName, cpName, migNs, resetConfigPath string, extraArgs map[string]string) {
+	var migration *storkapi.Migration
+	err = storkctlcli.ScheduleStorkctlMigrationSched(migSchedName, cpName, migNs, extraArgs)
+	log.FailOnError(err, "Error creating migrationschedule: %v", err)
+	err = hardSetConfig(resetConfigPath)
+	log.FailOnError(err, "Error setting destination config: %v", err)
+	migSchedule, err := storkops.Instance().GetMigrationSchedule(migSchedName, migNs)
+	log.FailOnError(err, "failed to get migrationschedule %v, err: %v", migSchedName, err)
+	time.Sleep(time.Second * 30)
+	migrations := migSchedule.Status.Items["Interval"]
+	for _, mig := range migrations {
+		migration, err = storkops.Instance().GetMigration(mig.Name, migNs)
+		log.FailOnError(err, "failed to get migration for migrationschedule %v, err: %v", migSchedName, err)
+		err = WaitForMigration([]*storkapi.Migration{migration})
+		log.FailOnError(err, "Migration failed with error: %v", err)
+	}
+}
+
+func performFailoverFailback(foFbParams failoverFailbackParam) {
+	err, output := storkctlcli.PerformFailoverOrFailbackStlCli(foFbParams.action, foFbParams.failoverOrFailbackNs, foFbParams.migrationSchedName, foFbParams.skipSourceOp, foFbParams.extraArgsFailoverFailback)
+	log.FailOnError(err, "Error running perform %v: %v", foFbParams.action, err)
+	splitOutput := strings.Split(output, "\n")
+	prefix := fmt.Sprintf("To check %s status use the command : `", foFbParams.action)
+	getStatusCommand := strings.TrimSpace(strings.TrimPrefix(splitOutput[1], prefix))
+	getStatusCommand = strings.TrimSuffix(getStatusCommand, "`")
+	getStatusCmdArgs := strings.Split(getStatusCommand, " ")
+	// Extract the action Name from the command args
+	actionName := getStatusCmdArgs[3]
+	if foFbParams.action == "failover" {
+		err = storkctlcli.WaitForActionSuccessful(storkapi.ActionTypeFailover, actionName, foFbParams.failoverOrFailbackNs, foFbParams.configPath, Inst().GlobalScaleFactor)
+	} else {
+		err = storkctlcli.WaitForActionSuccessful(storkapi.ActionTypeFailback, actionName, foFbParams.failoverOrFailbackNs, foFbParams.configPath, Inst().GlobalScaleFactor)
+	}
+	log.FailOnError(err, "Error in performing %v: %v", foFbParams.action, err)
+	validatePodsRunning(foFbParams.action, foFbParams.single, foFbParams.includeNs, foFbParams.excludeNs, foFbParams.contexts) 
+}
+
+func validatePodsRunning(action string, single, includeNs, excludeNs bool, contexts []*scheduler.Context) {
```

Comment:
A switch case might be a better option with respect to code readibility here.

Reply:
Done.